core/src/main/scala/org/http4s/util/Renderable.scala

```diff
@@ -120,3 +123,18 @@ class StringWriter(size: Int = 64) extends Writer {
 
   def result(): String = sb.toString
 }
+/** [[Writer]] that will result in a `ByteVector`
+  * @param bv initial ByteVector`
+  */
+case class ByteVectorWriter(private var bv:ByteVector = ByteVector.empty) extends Writer {
+
+  override def append(s: String)       = { bv = bv ++ ByteVector(s.getBytes);this}
+  override def append(char: Char)      = { bv = bv ++ ByteVector(char.toByte);this}
+  override def append(float: Float)    = { bv = bv ++ ByteVector(float.toByte);this}
+  override def append(double: Double)  = { bv = bv ++ ByteVector(double.toByte);this}
+  override def append(int: Int)        = { bv = bv ++ ByteVector(int.toByte);this}
+  override def append(long: Long)      = { bv = bv ++ ByteVector(long.toByte);this}
```

Comment:
I'm thinking that `toByte` for these number types is not appropriate. I suspect we need to clarify the purpose of the `Writer` type a bit, but I bet these need to be `long.toString.getBytes` or something similar. That also means we need to have a `Charset` so that we know how to encode the `Char`s.